### PR TITLE
Unify Rsvp{Lookup, Footer}Page into RsvpPage

### LINKED ISSUE #159: Implement RSVP lookup UI to retrieve a guest's RSVP by name
- To retrieve the guest's RSVP, we have them enter their name first.
- If the guest's RSVP can't be found, let them know and ask them to try again or reach out to us. 
- If the guest's RSVP is found, show them the RSVP form, prepopulated with their data.

### PR DESCRIPTION
**Pull request by huy-vuong**
# Issues

- Closes #159.

# Testing

## Manual Procedure

**Make sure to set up your `.env` file!** It should be in the `web/` directory, and follow the format of the `.env.example` file in there.

**The Deploy Preview will not work!** I have not yet set up `.env` files for it, so RSVP Lookup will fail.

### RSVP Found

1. Start the website up: `cd web/ && npm start`
2. Open the RSVP page in your browser, which should show the RSVP Lookup: http://localhost:8080/rsvp
3. Fill out the form as follows (it's based on the one user who with RSVP data in the Deta Base instance) — inputs are _**not** case sensitive_:
    - **First name:** furst
    - **Last name:** lahst
4. Click the **Submit** button.
5. The page should transition to the RSVP Editor with the **First name** and **Last name** fields prepopulated.

### RSVP Not Found

1. Start the website up: `cd web/ && npm start`
2. Open the RSVP page in your browser, which should show the RSVP Lookup: http://localhost:8080/rsvp
3. Fill out the form as follows (it's based on the one user who with RSVP data in the Deta Base instance) — inputs are _**not** case sensitive_:
    - **First name:** The
    - **Last name:** Uninvited
4. Click the **Submit** button.
5. The page should stay on the RSVP Lookup, but now showing this error message: **Couldn't find your name. Did you enter it the way it looks on your invitation?**.

### CODE REVIEW

**Comment by YanZ777** [comment]
What's a name I can test with?

File: web/src/components/RsvpLookupForm/RsvpLookupForm.jsx [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
+   const onSubmit = async ({ firstName, lastName }) => {
+      setErrorMessage(null);
+      try {
+         const response = await axios.get(
+            `https://${process.env.SERVER_HOST}/rsvps/${lastName}/${firstName}`,
+         );
+         onRsvpFound(response.data);
+      } catch (e) {
+         setErrorMessage(
+            e.response.status === StatusCodes.NOT_FOUND
```
Comment (YanZ777) [inline web/src/components/RsvpLookupForm/RsvpLookupForm.jsx:24]: **suggestion:** You might want to handle things like where there's no status at all, such an empty response.

**Review by YanZ777 — COMMENTED** [review]
Initial comment for now, I wasn't able to fully test this, but that might have been me being a derp.

File: web/src/components/RsvpLookupForm/RsvpLookupForm.jsx (hunk 1 above)
Comment (huy-vuong) [reply web/src/components/RsvpLookupForm/RsvpLookupForm.jsx:24]: Yeah, that's fair. We shouldn't get empty data, but I guess if we did, I'd probably funnel it to the same `else` case as here with the `'Something is sad right now. Can you let us know?'` message.

File: web/src/pages/RsvpPage.test.jsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+   });
+   await user.type(lastNameTextbox, 'Simpson');
+
+   const submitButton = screen.getByRole('button', { name: 'Submit' });
+   await user.click(submitButton);
+
+   await waitFor(() => {
+      expect(screen.getByText('RSVP Lookup')).toBeInTheDocument();
+      expect(screen.queryByText('RSVP Editor')).not.toBeInTheDocument();
+   });
```
Comment (YanZ777) [inline web/src/pages/RsvpPage.test.jsx:100]: **suggestion:** Could you add a test to make sure the transition between the components works?

**Review by YanZ777 — CHANGES_REQUESTED** [review]
I'd like a test for the transition between the lookup form and the editor form.

File: web/src/pages/RsvpPage.test.jsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
```
Comment (huy-vuong) [inline web/src/pages/RsvpPage.test.jsx:50]: Oops, this should've been that the editor is in the document and the look is not. Copy-pasting code ftl. ;_;

File: web/src/pages/RsvpPage.test.jsx (hunk 2 above)
Comment (huy-vuong) [reply web/src/pages/RsvpPage.test.jsx:102]: *(Discussed in-person. Test is there (see other comment), but it's named badly and I also wrote the wrong `expect`s for it. Oops.)*

**Review by YanZ777 — APPROVED** [review]

**Pull request merged**